src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets

```diff
@@ -269,7 +269,7 @@ Copyright (c) .NET Foundation. All rights reserved.
       </PropertyGroup>
     </When>
     
-    <When Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(OutputType)' == 'Exe'">
+    <When Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(TargetExt)' == '.exe'">
```

Comment:
This is not the only place that's broken. Winexe's won't get binding redirects turned on or the right logic for native dependencies on .NET framework. We need to scrub.

There was already a proposal to have something like 'IsExecutable' get respected regardless of OutputType. So that test projects (which are dlls) could opt-in to behaviors like binding redirects without changing the output type. I wonder if we can kill two birds here.